readers/src/main/scala/com/salesforce/op/readers/DataReader.scala

```diff
@@ -179,21 +180,30 @@ trait DataReader[T] extends Reader[T] with ReaderKey[T] {
 
     rawData match {
       case Left(rdd) =>
-        val d = rdd.flatMap(record => generateRow(key(record), record, rawFeatures))
+        val d = rdd.flatMap(record => generateRow(key(record), record, rawFeatures, schema))
         spark.createDataFrame(d, schema)
       case Right(ds) =>
-        implicit val rowEnc = RowEncoder(schema)
-        ds.flatMap(record => generateRow(key(record), record, rawFeatures))
+        val inputSchema = ds.schema.fields
+         if (schema.forall(fn => inputSchema.exists(
```

Comment:
does this check here validates that the dataframe schema exactly matches the input features?

Reply:
it checks that the schema for all the input features matches that of the input dataframe features (there can be additional features in the dataframe). I will add a comment.